certbot-apache/certbot_apache/_internal/configurator.py

```diff
@@ -156,9 +157,9 @@ def _prepare_options(self):
                 self.options[o] = self.OS_DEFAULTS[o]
 
         # Special cases
-        self.options["version_cmd"][0] = self.option("ctl")
-        self.options["restart_cmd"][0] = self.option("ctl")
-        self.options["conftest_cmd"][0] = self.option("ctl")
+        cast(List[str], self.options["version_cmd"])[0] = self.option("ctl")
```

Comment:
To try to avoid as much cast as possible.

In fact I have an alternative here, way more elegant I think, but it requires to change the logic of the code, while initially I wanted to avoid it as soon as there is nothing wrong with the code itself. But it may be worth it still, I let appreciate it.

Instead of using a dict for `self.options`, what about defining a proper object like `ConfiguratorOptions` that would hold all parameters? On top of obviously removing the need for casts, because we will be able to define the type for each parameter, I think it would improve the code quality, since this dict hold a lot of parameters, that are critical for the good behavior of the configurator and its different overrides.

Reply:
I didn't look at all the ways options are used which I would need to do to really comment on the design, but I think something like that is a lot better than what we have now and is the kind of refactoring that I personally think probably makes sense as we do this `mypy` work. In some ways, we're doing it to satisfy `mypy`, but I also think it's a better design even if we weren't using `mypy`.

I don't personally think we should do that as part of this PR since I think what you have here works just fine, but if you wanted to do that, I do think it's a good use of time and I'd happily review the PR or look at the design more closely beforehand if you'd like.

Reply:
OK let's do this.  I will do that on a separate PR in a very near future.

Reply:
Great! Thanks for picking that up. Taking a look at how `self.options` is used, I think that should work pretty well.